src/models/api_key.rs

```diff
@@ -0,0 +1,27 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+pub struct ApiKeyRequest {
+    pub name: String,
+    pub permissions: Vec<String>,
+    pub allowed_origins: Option<Vec<String>>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ApiKeyModel {
+    pub id: String,
+    pub value: String,
```

Comment:
We should use SecretString to prevent memory leaks when using sensitive values.

For example it also takes care of preventing deserialisation with raw value (logs).